core/domain/question_domain.py

```diff
@@ -122,25 +140,53 @@ def to_dict(self):
             dict. A dict representation of the Question instance.
         """
         return {
-            'question_id': self.question_id,
+            'id': self.id,
             'question_data': self.question_data,
             'question_data_schema_version': self.question_data_schema_version,
-            'language_code': self.language_code
+            'language_code': self.language_code,
+            'status': self.status
         }
 
     def validate(self):
         """Validates the Question domain object before it is saved."""
 
-        if not isinstance(self.question_id, basestring):
+        if not isinstance(self.id, basestring):
             raise utils.ValidationError(
-                'Expected ID to be a string, received %s' % self.question_id)
+                'Expected ID to be a string, received %s' % self.id)
 
         if not isinstance(self.question_data, dict):
             raise utils.ValidationError(
                 'Expected question_data to be a dict, received %s' %
                 self.question_data)
 
-        at_least_one_correct_answer = False
+        question_data = exp_domain.State.from_dict(self.question_data)
+        question_data.validate({}, True)
+
+        if not isinstance(self.question_data_schema_version, int):
+            raise utils.ValidationError(
+                'Expected question_data_schema_version to be a integer,' +
+                'received %s' % self.question_data_schema_version)
+
+        if not isinstance(self.language_code, basestring):
+            raise utils.ValidationError(
+                'Expected language_code to be a string, received %s' %
+                self.language_code)
+
+        if not any([self.language_code == lc['code']
+                    for lc in constants.ALL_LANGUAGE_CODES]):
+            raise utils.ValidationError(
+                'Invalid language code: %s' % self.language_code)
+
+        if not isinstance(self.status, basestring):
+            raise utils.ValidationError(
+                'Expected status to be a string, received %s' %
+                self.status)
+
+    def validate_for_publishing_or_send_for_review(self):
```

Comment:
Not sure I like this -- can we restructure this to be similar to exploration.validate() where there is a `strict=True` option? I think that's cleaner and more consistent.

Reply:
Rather to check question_data (I renamed it to question_state_data), we can simply use State.validate() and self.validate() for other validation. Would be much cleaner